sklearn/preprocessing/_data.py

```diff
@@ -74,7 +74,7 @@ def _handle_zeros_in_scale(scale, copy=True):
         if copy:
             # New array to avoid side-effects
             scale = scale.copy()
-        scale[scale == 0.0] = 1.0
+        scale[np.isclose(scale, 0.0)] = 1.0
```

Comment:
Please add a [non-regression test](https://en.wikipedia.org/wiki/Non-regression_testing) that would fail at master but pass in this PR.

Reply:
Hi @thomasjpfan do you have a link or anything with an example showing me what you're looking for? This is my first pull request so it's all new to me. I have a script in issue #17794 highlighting the issue. Would modifying this to demonstrate the correction be sufficient, or is there something else I need to do?

Reply:
Adding a test similar to https://github.com/scikit-learn/scikit-learn/issues/17794 to make sure this fixes the issue will work.